### DISCUSSION TITLE
A question about transmission occurring

### DISCUSSION THREAD

**Discussion by DWin233** [fangq/mcx#211] [Help]
I'm reading the source code to learn VMC and have a little doubt.

In the device function `reflectray(...)`, when judging whether to enter the background medium, is there an implicit assumption that the upper medium is not the background medium? If so, I would like to understand why this assumption is made?

```   
if (((nuvox->sv.isupper) ? nuvox->sv.isupper : nuvox->sv.lower) == 0) { /*transmit to background medium*/
    return 1;
}
```

**Comment by fangq** [comment]
@DWin233, this if-branch is related to our [SVMC algorithm](https://www.osapublishing.org/boe/abstract.cfm?uri=boe-11-11-6262) implemented by @ShijieYan. In the SVMC volume data structure, one voxel can contain two labels - thus split-voxel - separated by an sub-voxel interface. the `sv.isupper` flag tells whether the current photon is located in the upper-label portion or the lower-label portion if the voxel does contain two labels.

the above statement is simply retrieving the "current" medium label and test if it is 0 - if in a single-medium voxel, `sv.isupper` will never be 1, therefore, `sv.lower` will always be used; if in a two-medium voxel, depends on sv.isupper, we retrieve the corresponding label.
